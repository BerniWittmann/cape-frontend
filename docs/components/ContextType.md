# ContextType

Renders a Single Context Type

## Props

<!-- @vuese:ContextType:props:start -->
|Name|Description|Type|Required|Default|
|---|---|---|---|---|
|contextType|The Context Type Object|`Object`|`true`|-|
|closable|Option whether the Context Type can be deleted or not|`Boolean`|`false`|false|

<!-- @vuese:ContextType:props:end -->


## Events

<!-- @vuese:ContextType:events:start -->
|Event Name|Description|Parameters|
|---|---|---|
|close|Fired when the close button of the Context Type was clicked|-|

<!-- @vuese:ContextType:events:end -->


