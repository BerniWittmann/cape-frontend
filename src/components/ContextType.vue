<template>
  <div class="context-type">
    <span class="tag tag-row__tag el-tag">
      <i class="fa fa-lg fa-fw" :class="contextType.icon" id="fa-icon"></i>
      {{ contextType.name }}
      <i class="el-tag__close el-icon-close" @click="emitClose" v-if="closable"></i>
    </span>
  </div>
</template>

<script>
/* ============
 * Context Type Component
 * ============
 *
 * A component to view a single Context Type
 */

export default {
  props: {
    contextType: {
      type: Object,
      required: true
    },

    closable: {
      type: Boolean,
      default: false
    }
  },

  computed: {},

  methods: {
    emitClose() {
      this.$emit('close')
    }
  }
}

</script>

<style lang="scss">

#fa-icon {
  font-size: 1.5em;
  margin-right: 5px;
}

</style>
