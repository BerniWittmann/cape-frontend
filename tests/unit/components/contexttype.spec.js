import { shallowMount } from '@vue/test-utils'
import { i18n } from '../setupPlugins'

import ContextType from '@/components/ContextType.vue'

// TODO add .toBe(true) to exists

describe('Components', () => {
  describe('ContextType', () => {
    let propsData
    let cmp

    beforeEach(() => {
      propsData = {
        contextType: {
          id: '142',
          name: 'My ContextType',
          icon: 'fa-heart'
        }
      }
      render()
    })

    function render() {
      cmp = shallowMount(ContextType, {
        i18n,
        propsData
      })
    }

    it('renders', () => {
      expect(cmp.html()).toMatchSnapshot()
      const contextTypeComponent = cmp.find('.context-type')
      expect(contextTypeComponent.exists()).toBeTruthy()
    })

    it('renders the name', () => {
      expect(cmp.text()).toEqual('My ContextType')
    })

    it('renders the icon', () => {
      expect(cmp.find('.context-type').exists()).toBeTruthy()
      expect(cmp.find('.fa').exists()).toBeTruthy()
      expect(cmp.find('.fa-lg').exists()).toBeTruthy()
      expect(cmp.find('.fa-heart').exists()).toBeTruthy()
    })

    it('can handle an undefined icon', () => {
      cmp.setProps({
        contextType: {
          id: '123',
          name: 'My ContextType'
        }
      })
      expect(cmp.find('.context-type').exists()).toBeTruthy()
      expect(cmp.find('.fa').exists()).toBeTruthy()
      expect(cmp.find('.fa-lg').exists()).toBeTruthy()
    })

    it('is not closeable by default', () => {
      const tagComponent = cmp.find('div')
      expect(tagComponent.exists()).toBeTruthy()
      expect(tagComponent.props('closable')).toEqual(false)
    })

    it('can be closable', () => {
      cmp.setProps({
        closable: true
      })
      expect(cmp.find('.el-icon-close').exists()).toBeTruthy()
      const tagComponent = cmp.find('div')
      expect(tagComponent.exists()).toBeTruthy()
      expect(tagComponent.props('closable')).toEqual(true)
    })

    it('emits a close event on close', () => {
      cmp.setProps({
        closable: true
      })
      expect(cmp.emitted().close).toBeFalsy()

      cmp.vm.emitClose()

      expect(cmp.emitted().close).toBeTruthy()
    })
  })
})
