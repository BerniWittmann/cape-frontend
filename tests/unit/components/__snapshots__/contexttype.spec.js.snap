// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components ContextType renders 1`] = `
<div class="context-type"><span class="tag tag-row__tag el-tag"><i id="fa-icon" class="fa fa-lg fa-heart"></i>
    My ContextType
    <!----></span></div>
`;
